README.md

```diff
@@ -12,4 +12,122 @@ The repository structure is as follows:
 - `Generation`: C++, Python and Z3 code which generates explicit colourings for certain sets
 - `Lean`: Lean formal proofs of the results
 - `Verso`: Lean code to generate the website
-- `site`: A partial Jekyll website, which completed by the code in `Verso`
\ No newline at end of file
+- `site`: A partial Jekyll website, which completed by the code in `Verso`
+
+## Prerequisites
+
+### For Lean Builds
+- [Elan](https://github.com/leanprover/elan) - Lean version manager
+  ```bash
+  curl https://elan.lean-lang.org/elan-init.sh -sSf | sh
+  ```
+
+### For Generation Code
+- A C++ compiler with OpenMP support (e.g., GCC)
+- Python 3 with dependencies from `requirements.txt`:
+  ```bash
+  pip install -r requirements.txt
+  ```
+
+### For Website Build
+- Ruby 3.1+ with Bundler
+- Jekyll (installed via Bundler)
+
+## Build Instructions
+
+### Building the Lean Proofs
+
+**IMPORTANT**: Before building, you must download the pre-compiled mathlib cache to avoid
+a 60+ minute build time.
+
+```bash
+cd Lean
+lake exe cache get  # Download mathlib cache (required!)
+lake build          # Build the project
```

Comment:
The Lean steps should happen after the Generation steps, since the lean build requires the colouring log files